### PR TITLE
[electron] Remove winston as a dependency

### PR DESCRIPTION
**Pull request by matheusd**
This removes the winston log library as a dependency from the project. This helps reduce the number of dependencies on the main electron layer.

This is done by implementing a very basic logger that replicates the features we were using from winston: writing both to a log file and to the terminal when `debug` is set.

Dependencies of the main electron layer before winston is removed:

electron-deps-winston

After removing winston:
electron-deps-no-winston

### CODE REVIEW

File: app/main.development.js [hunk 1]
```diff
@@ -256,14 +256,13 @@ checkAndInitWalletCfg(false);
...
-  process.versions.electron,
-  process.versions.chrome
+  `Versions: Decrediton: ${app.getVersion()}, ` +
+    `Electron: ${process.versions.electron}, ` +
+    `Chrome: ${process.versions.chrome}`
 );
 
 process.on("uncaughtException", (err) => {
-  logger.log("error", "UNCAUGHT EXCEPTION", err);
+  logger.log("error", "UNCAUGHT EXCEPTION: " + err);
```
Comment (amassarwi) [inline app/main.development.js:265]: string template

File: app/main_dev/launch.js [hunk 2]
```diff
@@ -700,7 +700,7 @@ export const launchDCRWallet = async (
           e &&
           e.code &&
           e.code != "EOF" &&
-          logger.log("error", "tx stream error", e)
+          logger.log("error", "tx stream error: " + e)
```
Comment (amassarwi) [inline app/main_dev/launch.js:703]: string template

File: app/main_dev/launch.js [hunk 3]
```diff
@@ -873,7 +872,7 @@ export const launchDCRLnd = (
       if (!lastDcrdErr || lastDcrdErr === "") {
         lastDcrdErr = lastPanicLine(GetDcrdLogs());
       }
-      logger.log("error", "dcrd closed due to an error: ", lastDcrdErr);
+      logger.log("error", "dcrd closed due to an error: " + lastDcrdErr);
```
Comment (amassarwi) [inline app/main_dev/launch.js:875]: string template

File: app/wallet/app.js [hunk 4]
```diff
@@ -63,16 +71,16 @@ const formatLogArgs = (msg, args) => {
...
+      return String(arg);
     } else {
       return JSON.stringify(arg);
     }
   };
 
-  const logMsg = args.reduce((a) => a + "%s ", "%s ");
-  const logArgs = [msg, ...args.map(formatArg)];
+  const logArgs = args.map(formatArg);
+  const logMsg = msg + " " + logArgs.join(" ");
```
Comment (amassarwi) [inline app/wallet/app.js:81]: string template

**Review by amassarwi — APPROVED** [review]
Nice work 🥇 
Left some minors

**Review by bgptr — CHANGES_REQUESTED** [review]
The code looks good and works fine.
I found only one critical issue: when I try to navigate to `Logs` view the application quits with this error message:

``` 2021-07-21 21:44:22.502 [ERR] DCTN: UNCAUGHT EXCEPTION: TypeError: Cannot read property 'find' of undefined
/home/bgptr/go/src/github.com/decred/decrediton/app/main.js:729
_fs.default.existsSync(walletsDirectory)||_fs.default.mkdirSync(walletsDirectory,{recursive:true});_fs.default.existsSync(mainnetWalletsPath)||_fs.default.mkdirSync(mainnetWalletsPath,{recursive:true});_fs.default.existsSync(testnetWalletsPath)||_fs.default.mkdirSync(testnetWalletsPath,{recursive:true});(0,_paths.checkAndInitWalletCfg)(true);(0,_paths.checkAndInitWalletCfg)(false);logger.log("info","Using config/data from:"+_electron.app.getPath("userData"));logger.log("info",`Versions: Decrediton: ${_electron.app.getVersion()}, `+`Electron: ${process.versions.electron}, `+`Chrome: ${process.versions.chrome}`);process.on("uncaughtException",err=>{logger.log("error",`UNCAUGHT EXCEPTION: ${err}`);throw err;});const installExtensions=async()=>{if(true){const extensions=[_electronDevtoolsInstaller.REACT_DEVELOPER_TOOLS,_electronDevtoolsInstaller.REDUX_DEVTOOLS];const forceDownload=!!process.env.UPGRADE_EXTENSIONS;for(const name of extensions){try{await(0,_electronDevtoolsInstaller.default)(name,forceDownload);}catch(e){console.log("Error installing extension: "+e);}}}};const{ipcMain}=__webpack_require__(/*! electron */"electron");// handleEvent listens on the given channel for ipcRenderer.invoke() calls and


                                                                  ^

TypeError: Cannot read property 'find' of undefined
    at IpcMainImpl.<anonymous> (/home/bgptr/go/src/github.com/decred/decrediton/app/main.js:8821:42)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:10351)
    at Object.emit (events.js:315:20)
```

**Comment by matheusd** [comment]
Nice catch! Updated.

**Review by bgptr — APPROVED** [review]

**Review by alexlyp — APPROVED** [review]

**Pull request merged**